### DISCUSSION TITLE
zh/guide/advanced/meta

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#37] [Announcements]
# zh/guide/advanced/meta

AList v3的文档

https://alist.nn.ci/zh/guide/advanced/meta.html

**Comment by qian9008** [comment]
隐藏所有文件怎么表达？ 
  *                  ？

**Reply by FantasyFish** [reply]
如果是不想让访客看到你的文件，可以把访客账号guest的基本路径改为一个空目录，访客浏览就只会看到这个空目录了

**Comment by jxl3333** [comment]
每行一个正则表达式??是什么意思  怎么隐藏啊

**Comment by zzerosama** [comment]
没有像V2那样只显示某个文件夹了吗？

**Comment by zzerosama** [comment]
> 隐藏所有文件怎么表达？ 
>   *                  ？

直接复制总的文件夹名称就隐藏了，路径要选上一级

**Comment by zzerosama** [comment]
> 每行一个正则表达式??是什么意思  怎么隐藏啊

就是每行放一个文件夹名字，回车隔开，V2版本是英文逗号，这个V3是转行，说正则表达式是因为作者想要设定一个表达式隐藏多个类似的文件夹吧，一般就每行复制一个要隐藏的文件夹名称就行，就会识别设定路径下隐藏的文件夹或文件

**Comment by yws112358** [comment]
这个元信息添加的密码不会影响下级目录，比如别人有下级目录的链接，后期上级目录加密码了，仍然可以直接访问下级目录。

**Reply by xingyu42** [reply]
那就应用到全部文件夹，如果哪个文件夹要分享就添加一个元数据没密码可以覆盖过去

**Comment by NKU100** [comment]
我想仅显示一个Onedrive，于是采用了以下的写法

```
^(?!Onedrive).*$
```

但是报错提示

> ^(?!Onedrive).*$ is illegal: error parsing regexp: invalid or unsupported Perl syntax: `(?!`

请问是我的写法不对，还是没有支持零宽断言？

**Comment by PanDaDa0202** [comment]
Database is locked

**Comment by nelson714** [comment]
文件夹名字中间有空格是不是不可以隐藏，我尝试过用%20也不行。可不可以说明一下我们可以用的语法？

**Comment by iwsbiwy** [comment]
这样设置根目录密码，为什么直接输入网址（不登陆账号），还能看到所有文件夹。
[![pSn9NP1.png](https://s1.ax1x.com/2023/01/11/pSn9NP1.png)](https://imgse.com/i/pSn9NP1)
[![pSn9BrD.png](https://s1.ax1x.com/2023/01/11/pSn9BrD.png)](https://imgse.com/i/pSn9BrD)

**Reply by 17yuite** [reply]
V2版本：设置——后端——检查父文件密码

**Reply by 17yuite** [reply]
这样就算知道上级目录结构也会提示输入密码

**Comment by backtracker** [comment]
这个隐藏到底该怎么用？试了N次了都是不行

**Reply by raydoomed** [reply]
^文件夹名$

**Comment by raydoomed** [comment]
隐藏表达：^文件夹名$

**Comment by lianyanjiajia** [comment]
根目录设置密码 访客还是能正常访问

**Comment by lianyanjiajia** [comment]
感谢，这样可以了



---Original---
From: ***@***.***&gt;
Date: Thu, Feb 2, 2023 00:11 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/advanced/meta (Discussion #37)




 
如果是不想让访客看到你的文件，可以把访客账号guest的基本路径改为一个空目录，访客浏览就只会看到这个空目录了

**Reply by firgk** [reply]
把 游客的 无需密码访问 关了 设置密码就可以生效了

**Comment by vipshihua** [comment]
隐藏功能折腾了半天的人现身说法
使用的是V3版本，元信息-路径栏输一个/就好，表示根目录；隐藏栏这里填入文件夹名称就行，不需要^和$等其它符号。
要注意的是保存完了得退出登入，以游客视角来查看~ 我折腾了半天就是因为账号登着怎么设置都看的到~

**Comment by xiyuyeyun** [comment]
试了n次不行，看评论也是一片搞不定，一个加密隐藏功能弄得这么复杂意义何在？
有没有大佬出个图文教程？

**Reply by firgk** [reply]
把 游客的 无需密码访问 关了 设置密码就可以生效了

**Comment by Jerry-FaGe** [comment]
没懂这个`写入`是什么意思，如果我不给`guest`用户上传权限那即使开着这个开关访客也上传不了；如果我给了上传权限那所有没有元信息的文件夹访客就都能上传

**Comment by firgk** [comment]
把 游客的  **无需密码访问** 关了 设置密码就可以生效了

--我也折腾了十几分钟

**Comment by igbh** [comment]
怎么在alist的首页加载readme.md？？？
找了好久都没明白。。。

**Comment by firgk** [comment]
Sorry, I haven't used it before. I don't use it much now&nbsp;


IQOO
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年3月14日(星期二) 中午11:31
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/advanced/meta (Discussion #37)





 
怎么在alist的首页加载readme.md？？？
 找了好久都没明白。。。

**Reply by igbh** [reply]
呃？啥？你没用过？你是作者吗？

**Comment by firgk** [comment]
我只是用户（only user）
这个也许是你在找的
https://www.bilibili.com/video/BV16v4y1v7Ln/?vd_source=a9dde2385c9a6de7816fb999b5b76c10
&nbsp;


IQOO
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年3月14日(星期二) 中午11:51
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/advanced/meta (Discussion #37)





 
呃？啥？你没用过？你是作者吗？

**Comment by qwwzi** [comment]
为什么我已经把无需密码访问取消了，访客还能看？目前最新的版本  是bug吗？

**Reply by zzerosama** [reply]
编辑guest账号，下边停用勾上

**Comment by Jeamorg** [comment]
隐藏教程

将在根目录下的 `test` 和 `example` 文件夹隐藏

在元信息里面添加：
 - 途径填
 /
 - 隐藏填
 ^Test$
^Music$

此时 `test` 和 `example` 文件夹已经被隐藏起来了，但是没有密码只能登录后访问

为 `test` 文件夹添加密码：

在元信息里面添加：
- 途径填
 /test
 - 密码填
123

此时 `test` 文件夹可以通过密码访问了

**Comment by cancundeyingzi** [comment]
说明很不方便，，，不能用.d/123456/README.md这种链接写法

**Reply by cancundeyingzi** [reply]
就是所谓的内部链接

**Comment by Mooncan** [comment]
文章里已经说了，此处的正则是```Golang```实现的，所以匹配的正则规则肯定用的是```Golang```的规则，参考[此处](http://c.biancheng.net/view/5124.html)写一下正则规则即可。

**Comment by Perter1999** [comment]
请问下，如何将1个文件隐藏后，连管理员在前台也看不到？只能后台用命令行看？

**Comment by sea-ten** [comment]
请问一下如果是自用，没有启用访客，是不是不会存在被爬虫等风险？
